src/test/groovy/graphql/execution/NonNullableFieldValidatorTest.groovy

```diff
@@ -7,7 +7,9 @@ import static graphql.schema.GraphQLNonNull.nonNull
 
 class NonNullableFieldValidatorTest extends Specification {
 
-    ExecutionContext context = Mock(ExecutionContext)
+    ExecutionContext context = Mock(ExecutionContext) {
+        propagateErrors() >> true
+    }
 
     def parameters = Mock(ExecutionStrategyParameters) {
         getPath() >> ResultPath.rootPath()
```

Comment:
Can we please get two types of tests in place.

A unit test showing NonNullableFieldValidator working (not throwing exceptions say)

And a end to end integration test with the query directives in place and a DF returning null on a non null type

Reply:
Tests added, they found [an issue](https://github.com/graphql-java/graphql-java/pull/3772/commits/eb00f570ac12b8bf06eea43d0344d12b5305516d#diff-d839ec58e4dfc74c0cc52918ee6471da852cfcfdb5422422dc2e5561993470a9R61) already if the resolver returned `null` and did not throw